src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -988,6 +1027,24 @@ void IHttp2StreamLifetimeHandler.OnStreamCompleted(int streamId)
             }
         }
 
+        private void StartKeepAliveTimeout()
+        {
+            _context.TimeoutControl.SetTimeout(ServerOptions.Limits.KeepAliveTimeout.Ticks, TimeoutReason.KeepAlive);
+            _currentTimeout = TimeoutReason.KeepAlive;
+        }
+
+        private void StartRequestHeadersTimeout()
+        {
+            _context.TimeoutControl.SetTimeout(ServerOptions.Limits.RequestHeadersTimeout.Ticks, TimeoutReason.RequestHeaders);
+            _currentTimeout = TimeoutReason.RequestHeaders;
+        }
+
+        private void CancelNonRateTimeout()
```

Comment:
Do you expect a matching CancelRateTimeout?

Reply:
Not sure actually. I think I'm going to take @JunTaoLuo's recommendation and just put a CurrentSetTimeoutReason on TimeoutControl and remove these methods for now.